### DISCUSSION TITLE
Allow assignees option to be a github team

### DISCUSSION THREAD

**Discussion by GuillaumeRahbari** [renovatebot/renovate#24804] [Suggest an Idea]
**What would you like Renovate to be able to do?**

I would like Renovate to take a Github team as a possible value for the [assignees](https://docs.renovatebot.com/configuration-options/#assignees) option.

**Did you already have any implementation ideas?**

I did not dig into the code but I was wondering if it was possible for renovate to "learn" or to "list" the people who are in a github team ? This way Renovate could generate the assignees list from the github team.

**Are there any workarounds or alternative ideas you've tried to avoid needing this feature?**
For now the workaround is to list all the github team members in the assignees list option. But it means that this option need to be updated whenever the github team members changes. Also, whenever a teammate is on vacation, we are able to change the github team settings in order to avoid the teammate to be assigned to pull requests for the time he is on vacation. It means that we also need to update our renovate config to update the list 

**Is this a feature you'd be interested in implementing yourself?**

Not really, sorry

**Comment by rarkins** [comment]
Notes:
- Part of the need for this is because you can't assign a team directly in GitHub
- It needs to be compatible with `assigneesSampleSize`, which is implemented in the worker layer, so probably we should implement this logic/awareness in the worker layer too

Idea:
- A new optional function in `platform` called `getTeamMembers()` that takes a team name as argument and returns a list of usernames
- Renovate can call this if it is given a `team:_________` value for `assignees`, and substitute the return value

**Comment by rarkins** [comment]
@JamieMagee anything you'd add here?

**Comment by JamieMagee** [comment]
@rarkins I think we can assign a team directly, but it's a different parameter than assigning regular reviewers. [Request reviewers for a pull request](https://docs.github.com/en/free-pro-team@latest/rest/reference/pulls#request-reviewers-for-a-pull-request) shows that we can set `team_reviewers` instead of `reviewers` and it should work™

@GuillaumeRahbari as a workaround, you should be able to use a [CODEOWNERS file](https://docs.github.com/en/free-pro-team@latest/github/creating-cloning-and-archiving-repositories/about-code-owners) to assign teams of reviewers

There's also an overlap in Renovate's CODEOWNERS support, which brings CODEOWNERS support to platforms that don't support it natively (Gitea & BitBucket). So, I see 2 work items from this request:
1. Add support for `team:____` in `assignees`
2. Support teams in CODEOWNERS

**Comment by GuillaumeRahbari** [comment]
Hi @JamieMagee,

I'm not really sure to understand the workaround. I added the CODEOWNERS file in my test repository. But when a PR is raised and the pipeline is failing, no one is assigned as "assignee" (not "reviewers") from the team. And I cannot manually select a team in the assignee section of a Github PR request.

For now my workaround has been to declare a list of all members of the github team in the renovate config file (but not ideal).

**Comment by JamieMagee** [comment]
@GuillaumeRahbari Here's what I think is the key excerpt from the docs:

> A CODEOWNERS file uses a pattern that follows the same rules used in gitignore files. The pattern is followed by one or more GitHub usernames or team names using the standard `@username` or `@org/team-name` format.

Are you using the correct format for your team name?

It does sound odd that you can't even manually assign the team. Can you read [this help page](https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-organizations-and-teams/managing-code-review-assignment-for-your-team) and see if you _could_ set it automatically? If not, I recommend reaching out to GitHub support.

**Comment by GuillaumeRahbari** [comment]
@JamieMagee I guess I use the correct name because it assigned people as "reviewers" on every pull request. As I understand the help page you provided, CODEOWNERS is a way to only add reviewers automatically in a github PR. But what I would like is to assign automatically "assignees" and not "reviewers" (and I don't see a mention of it in the github documentation, that's why I think it is not something github allows)

**Comment by rarkins** [comment]
@JamieMagee I think based on the description that you can't assign a team, only add them as reviewer? [ref](https://github.com/dear-github/dear-github/issues/170)

So the idea of this feature request is to let Renovate users pick a team name for assigning but Renovate will "expand" the team name into usernames and add them to the actual PR (taking into account `assigneeSampleSize` too if configured)

**Comment by JamieMagee** [comment]
Ah yes, you're correct. I've gotten assignees and reviewers mixed up 🤦

**Comment by Wirone** [comment]
FYI in Gitlab it does not work too:

```
 INFO: PR created (repository=foo/bar, branch=feature/renovate/aws-aws-sdk-php-3.x-lockfile)
       "pr": 7922,
       "prTitle": "fix(composer): update dependency aws/aws-sdk-php to v3.215.1"
 WARN: Failed to add assignees (repository=foo/bar, branch=feature/renovate/aws-aws-sdk-php-3.x-lockfile)
       "iid": 7922,
       "assignees": ["team/composer"]
 WARN: Failed to get IDs of the new reviewers (repository=foo/bar, branch=feature/renovate/aws-aws-sdk-php-3.x-lockfile)
       "err": {
         "message": "Cannot read property 'id' of undefined",
         "stack": "TypeError: Cannot read property 'id' of undefined\n    at getUserID (/usr/src/app/node_modules/renovate/lib/modules/platform/gitlab/http.ts:10:13)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)\n    at /usr/src/app/node_modules/p-map/index.js:57:22"
       }
```

We use Gitlab groups as nested organisational structure, teams are assigned as code owners. It would be great if MR reviewers could be set using the same approach.

**Comment by viceice** [comment]
@Wirone You have a different issue, please open a new feature request, currently we only allow users, so it's failing here:

https://github.com/renovatebot/renovate/blob/dca3418bbdbf1024ac6807e591562b4042bc846d/lib/modules/platform/gitlab/http.ts#L7-L11

Please also provides API references we need to call for teams and how we can easily distingisch between user and team / group.
